From b6f40b52d2d02913732189f37effd7652952efb2 Mon Sep 17 00:00:00 2001
From: Norbert Pfeiler <norbert.pfeiler+git@gmail.com>
Date: Tue, 12 May 2020 17:09:58 +0200
Subject: [PATCH] remove html preprocessing
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

it breaks the windows build with GCC 10
which doesn’t like proper apostrophes in c++ identifiers (the html text)

it wouldn’t even error if an incorrectly spelled macro was used anyway
---
 CMakeLists.txt                                   | 2 +-
 resources/cheatsheet/mergemode.html.in           | 8 +++-----
 resources/cheatsheet/mergetracingmode.html.in    | 8 +++-----
 resources/cheatsheet/paintmode.html.in           | 6 ++----
 resources/cheatsheet/reviewmode.html.in          | 8 +++-----
 resources/cheatsheet/tracingadvancedmode.html.in | 8 +++-----
 resources/cheatsheet/tracingmode.html.in         | 8 +++-----
 7 files changed, 18 insertions(+), 30 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2dd76e7c..4e56c312 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,7 +130,7 @@ file(GLOB cheatsheet_pages resources/cheatsheet/*.html.in)
 foreach(page ${cheatsheet_pages})
     get_filename_component(page_name ${page} NAME_WE) #name without path or extension
     add_custom_command(OUTPUT ${cheatsheet_dir}/${page_name}.html
-        COMMAND "${CMAKE_CXX_COMPILER}" -E -x c++ -P ${page} -I ${CMAKE_CURRENT_SOURCE_DIR} -o ${cheatsheet_dir}/${page_name}.html # -E: only preprocess, -x: language, -P: omit linemarkers
+        COMMAND ${CMAKE_COMMAND} -E copy ${page} ${cheatsheet_dir}/${page_name}.html
         COMMENT "Preprocessing ${page}"
         DEPENDS ${page} cheatsheet_dir_target
     )
diff --git a/resources/cheatsheet/mergemode.html.in b/resources/cheatsheet/mergemode.html.in
index 5fcdd153..6f443a65 100644
--- a/resources/cheatsheet/mergemode.html.in
+++ b/resources/cheatsheet/mergemode.html.in
@@ -1,5 +1,3 @@
-#include "htmlmacros.h"
-
 <html>
 <head>
 	<meta content="text/html; charset=utf-8"/>
@@ -9,13 +7,13 @@
 <body>
 	<h1>Segmentation Merge</h1>
 
-	<p><a href=ANNOTATION_SEG>Segmentation table view</a> | <a href=PREF_SEG>Visibility preferences</a></p>
-	<p><a href=NAVIGATION>Navigation preferences</a></p>
+	<p><a href=#annotation_seg>Segmentation table view</a> | <a href=#pref_seg>Visibility preferences</a></p>
+	<p><a href=#navi>Navigation preferences</a></p>
 
 	<p>
 	Merging is a segmentation method suitable for working on a machine-generated over-segmentation, i.e. the task consists of merging incorrectly splitted objects back together.
 	</p>
-	
+
 	<p>
 	<table>
 		<thead>
diff --git a/resources/cheatsheet/mergetracingmode.html.in b/resources/cheatsheet/mergetracingmode.html.in
index 8aa4d575..23494e1d 100644
--- a/resources/cheatsheet/mergetracingmode.html.in
+++ b/resources/cheatsheet/mergetracingmode.html.in
@@ -1,5 +1,3 @@
-#include "htmlmacros.h"
-
 <html>
 <head>
 	<meta content="text/html; charset=utf-8"/>
@@ -9,9 +7,9 @@
 <body>
 	<h1>Merge Tracing</h1>
 
-	<p><a href=ANNOTATION_SKEL>Skeleton table view</a> | <a href=PREF_NODE>Node preferences</a> | <a href=PREF_TREE>Tree preferences</a></p>
-	<p><a href=ANNOTATION_SEG>Segmentation table view</a> | <a href=PREF_SEG>Visibility preferences</a></p>
-	<p><a href=NAVIGATION>Navigation preferences</a></p>
+	<p><a href=#annotation_skel>Skeleton table view</a> | <a href=#pref_node>Node preferences</a> | <a href=#pref_tree>Tree preferences</a></p>
+	<p><a href=#annotation_seg>Segmentation table view</a> | <a href=#pref_seg>Visibility preferences</a></p>
+	<p><a href=#navi>Navigation preferences</a></p>
 
 	<p>
 	In Merge Tracing you perform a regular skeletonization, but each tree is attached to a designated Segmentation Object. As nodes are placed, the Objects beneath them are merged into the tree’s Object.
diff --git a/resources/cheatsheet/paintmode.html.in b/resources/cheatsheet/paintmode.html.in
index 3665e100..454f1c49 100644
--- a/resources/cheatsheet/paintmode.html.in
+++ b/resources/cheatsheet/paintmode.html.in
@@ -1,5 +1,3 @@
-#include "htmlmacros.h"
-
 <html>
 <head>
 	<meta content="text/html; charset=utf-8"/>
@@ -9,8 +7,8 @@
 <body>
 	<h1>Segmentation Paint</h1>
 
-	<p><a href=ANNOTATION_SEG>Segmentation table view</a> | <a href=PREF_SEG>Visibility preferences</a></p>
-	<p><a href=NAVIGATION>Navigation preferences</a></p>
+	<p><a href=#annotation_seg>Segmentation table view</a> | <a href=#pref_seg>Visibility preferences</a></p>
+	<p><a href=#navi>Navigation preferences</a></p>
 
 	<p>
 	Painting is a slower form of segmentation but enables you to perform fine-grained corrections by selecting an Object and bringing it into shape with painting and erasing.
diff --git a/resources/cheatsheet/reviewmode.html.in b/resources/cheatsheet/reviewmode.html.in
index 874d5c87..73d35c5b 100644
--- a/resources/cheatsheet/reviewmode.html.in
+++ b/resources/cheatsheet/reviewmode.html.in
@@ -1,5 +1,3 @@
-#include "htmlmacros.h"
-
 <html>
 <head>
 	<meta content="text/html; charset=utf-8"/>
@@ -9,9 +7,9 @@
 <body>
 	<h1>Review</h1>
 
-	<p><a href=ANNOTATION_SKEL>Skeleton table view</a> | <a href=PREF_NODE>Node preferences</a> | <a href=PREF_TREE>Tree preferences</a></p>
-	<p><a href=ANNOTATION_SEG>Segmentation table view</a> | <a href=PREF_SEG>Visibility preferences</a></p>
-	<p><a href=NAVIGATION>Navigation preferences</a></p>
+	<p><a href=#annotation_skel>Skeleton table view</a> | <a href=#pref_node>Node preferences</a> | <a href=#pref_tree>Tree preferences</a></p>
+	<p><a href=#annotation_seg>Segmentation table view</a> | <a href=#pref_seg>Visibility preferences</a></p>
+	<p><a href=#navi>Navigation preferences</a></p>
 
 	<p>
 	Review mode is intended for easy browsing of your segmentation and skeletonization without accidentally changing them.
diff --git a/resources/cheatsheet/tracingadvancedmode.html.in b/resources/cheatsheet/tracingadvancedmode.html.in
index 229a6889..3f165db8 100644
--- a/resources/cheatsheet/tracingadvancedmode.html.in
+++ b/resources/cheatsheet/tracingadvancedmode.html.in
@@ -1,5 +1,3 @@
-#include "htmlmacros.h"
-
 <html>
 <head>
 	<meta content="text/html; charset=utf-8"/>
@@ -9,9 +7,9 @@
 <body>
 	<h1>Tracing Advanced</h1>
 
-	<p><a href=ANNOTATION_SKEL>Skeleton table view</a></p>
-	<p><a href=PREF_NODE>Node preferences</a> | <a href=PREF_TREE>Tree preferences</a></p>
-	<p><a href=NAVIGATION>Navigation preferences</a></p>
+	<p><a href=#annotation_skel>Skeleton table view</a></p>
+	<p><a href=#pref_node>Node preferences</a> | <a href=#pref_tree>Tree preferences</a></p>
+	<p><a href=#navi>Navigation preferences</a></p>
 
 	<p>
 	Tracing Advanced offers more functionality than Tracing mode: Unlinked nodes and new trees can be created. This also enables you to make synapse annotations.
diff --git a/resources/cheatsheet/tracingmode.html.in b/resources/cheatsheet/tracingmode.html.in
index d5200e90..9cd390b2 100644
--- a/resources/cheatsheet/tracingmode.html.in
+++ b/resources/cheatsheet/tracingmode.html.in
@@ -1,5 +1,3 @@
-#include "htmlmacros.h"
-
 <html>
 <head>
 	<meta content="text/html; charset=utf-8"/>
@@ -9,9 +7,9 @@
 <body>
 	<h1>Tracing</h1>
 
-	<p><a href=ANNOTATION_SKEL>Skeleton table view</a></p>
-	<p><a href=PREF_NODE>Node preferences</a> | <a href=PREF_TREE>Tree preferences</a></p>
-	<p><a href=NAVIGATION>Navigation preferences</a></p>
+	<p><a href=#annotation_skel>Skeleton table view</a></p>
+	<p><a href=#pref_node>Node preferences</a> | <a href=#pref_tree>Tree preferences</a></p>
+	<p><a href=#navi>Navigation preferences</a></p>
 
 	<p>
 	Tracing mode is a simple skeletonization method in which you work on a single tree with connected nodes.
